src/data/matrix.lean

```diff
@@ -179,6 +179,24 @@ theorem diagonal_mul_diagonal [decidable_eq n] (d₁ d₂ : n → α) :
   diagonal d₁ * diagonal d₂ = diagonal (λ i, d₁ i * d₂ i) :=
 diagonal_mul_diagonal' _ _
 
+lemma is_add_monoid_hom_mul_left (M : matrix l m α) : 
+  is_add_monoid_hom (λ x : matrix m n α, M ⬝ x) :=
+{ to_is_add_hom := ⟨matrix.mul_add _⟩, map_zero := matrix.mul_zero _ }
+
+def is_add_monoid_hom_mul_right (M : matrix m n α) : 
+  is_add_monoid_hom (λ x : matrix l m α, x ⬝ M) :=
+{ to_is_add_hom := ⟨λ _ _, matrix.add_mul _ _ _⟩, map_zero := matrix.zero_mul _ }
+
+lemma sum_mul {β : Type*} (s : finset β) (f : β → matrix l m α)
+  (M : matrix m n α) : s.sum f ⬝ M = s.sum (λ a, f a ⬝ M) :=
+(@finset.sum_hom _ _ _ s f _ _ (λ x, x ⬝ M) 
+  (id (@is_add_monoid_hom_mul_right l _ _ _ _ _ _ _ M) : _)).symm
```

Comment:
Something funny is going on here. Without the `id` and the `: _` it doesn't type check. Something about `add_comm_monoid.to_add_monoid` not being equal to `matrix.add_monoid`. I couldn't get to the bottom of it.

Reply:
Yeah, this is strange. I don't know what's going on. This works, but using `exact` or `refine` instead of `convert` doesn't:
```lean
eq.symm $ @finset.sum_hom _ _ _ s f _ _ (λ x, x ⬝ M) $
  by convert is_add_monoid_hom_mul_right M
```

Should we merge this or were you planning to investigate more? (If we merge as-is, maybe leave a comment pointing out that there's something funny here.)

Reply:
I added a comment.